gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -70,6 +74,17 @@ public void apply(Project project) {
                         .apply(MetricSchemaMarkdownPlugin.class));
     }
 
+    private static boolean hasInjectionFramework(Project project) {
+        Configuration compileClasspath = project.getConfigurations().getByName("compileClasspath");
```

Comment:
```suggestion
        SourceSet main = project.getExtensions()
                .getByType(JavaPluginExtension.class)
                .getSourceSets()
                .getByName(SourceSet.MAIN_SOURCE_SET_NAME)
        Configuration compileClasspath = project.getConfigurations()
                .getByName(main.getCompileClasspathConfigurationName());
```

Reply:
If we want to use that notation should I also change https://github.com/palantir/metric-schema/blob/develop/gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java#L78 ?